### ISSUE TITLE
The gradient of `i0` for input 0 should be 0

### ISSUE THREAD

**Issue by Doraehhh** [google/jax#11479]
Based on the definition of `i0`, its gradient for input 0 should be 0. It is differentiable at the point 0, unlike `abs`

```py
import jax

a = jax.numpy.array([0.])
print(jax.jacrev(jax.numpy.i0)(a))
# [[1.]]
# It should be 0
```

**Comment by Doraehhh** [comment]
any update of this bug?

**Comment by jakevdp** [comment]
Thanks - sorry this fell through the cracks. It looks like this could be fixed by adding a custom JVP rule to `jax.numpy.i0`. Is this something you'd be interested in contributing? If not we can have someone on the team take care of it.
